js/canvas.js

```diff
@@ -0,0 +1,367 @@
+import { arrayOfRectangles } from './rectangles.js';
+import { canvasWindowIndent, startedRectangleFillColor, redColorInHex, stickyDistance } from './shared/constants.js';
+
+export class Canvas {
+  constructor(selectedCanvas) {
+    this.canvas = selectedCanvas;
+    this.ctx = this.canvas.getContext('2d');
+    this.draggingRect;
+    this.isDragOk = false;
+    this.offsetX;
+    this.offsetY;
+  }
+
+  start() {
+    this.setSize();
+    this.setOffsets_X_and_Y();
+    this.drawRectangles();
+    this.canvas.addEventListener('mousedown', this.mouseDownHandler.bind(this));
+    this.canvas.addEventListener('mousemove', this.mouseMoveHandler.bind(this));
+    this.canvas.addEventListener('mouseup', this.mouseUpHandler.bind(this));
+  }
+
+  setSize() {
+    const { innerWidth: width, innerHeight: height } = window;
+
+    this.canvas.width = width - canvasWindowIndent;
+    this.canvas.height = height - canvasWindowIndent;
+  }
+
+  setOffsets_X_and_Y() {
+    const domRect = canvas.getBoundingClientRect();
+    const { left, top } = domRect;
+
+    this.offsetX = left;
+    this.offsetY = top;
+  }
+
+  drawRectangles() {
+    this.clear();
+
+    arrayOfRectangles.forEach(rect => {
+      if (rect.isSticked && !rect.isCrossed) {
+        const { stickedX, stickedY, width, height } = rect;
+
+        this.ctx.fillStyle = rect.fillColor;
+        this.ctx.fillRect(stickedX, stickedY, width, height);
+      } else {
+        const { x, y, width, height } = rect;
+
+        this.ctx.fillStyle = rect.fillColor;
+        this.ctx.fillRect(x, y, width, height);
+      }
+    });
+  }
+
+  clear() {
+    const { width, height } = this.canvas;
+
+    this.ctx.clearRect(0, 0, width, height);
+  }
+
+  isRectangle_a_and_b_crossed(rectA, rectB) {
+    const topSideOfRectA = rectA.y;
+    const leftSideOfRectA = rectA.x;
+    const rightSideOfRectA = rectA.x + rectA.width;
+    const bottomSideOfRectA = rectA.y + rectA.height;
+    const topSideOfRectB = rectB.y;
+    const leftSideOfRectB = rectB.x;
+    const rightSideOfRectB = rectB.x + rectB.width;
+    const bottomSideOfRectB = rectB.y + rectB.height;
+
+    if (bottomSideOfRectA <= topSideOfRectB || rightSideOfRectA <= leftSideOfRectB) {
+      return false;
+    }
+
+    if (topSideOfRectA >= bottomSideOfRectB || rightSideOfRectB <= leftSideOfRectA) {
+      return false;
+    }
+
+    return true;
```

Comment:
lines 76-80 can be simplify - `return !(topSideOfRectA >= bottomSideOfRectB || rightSideOfRectB <= leftSideOfRectA);`